tests/functional/test_model_completeness.py

```diff
@@ -46,3 +48,48 @@ def test_paginators_and_waiters_are_not_lost_in_new_version(
             raise AssertionError(
                 f"{type_name} must exist for {service_name}: {e}"
             )
+
+
+def _endpoint_rule_set_cases():
+    for service_name in Session().get_available_services():
+        versions = Loader().list_api_versions(service_name, 'service-2')
+        for version in versions:
+            yield service_name, version
+
+
+@pytest.mark.parametrize(
+    "service_name, version",
+    _endpoint_rule_set_cases(),
+)
+def test_all_endpoint_rule_sets_exist(service_name, version):
+    """Tests the existence of endpoint-rule-set.json for each service
+    and verifies that content is present."""
+    loader = Loader()
```

Comment:
Nit: Since this test is effectively running in a loop, this creates a lot of `Loader` objects even though only a single one is needed. You can avoid that by just having a global (to this file) loader. We had a similar discussion in https://github.com/boto/botocore/pull/2737#discussion_r951764838. You won't get any caching benefits from consolidating into a single loader, but you'll save a few lines of code and a tiny amount of memory :)